lib/v2/screens/profile_screens/profile/components/citizenship_card.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:seeds/v2/constants/app_colors.dart';
+import 'package:seeds/v2/datasource/remote/model/profile_model.dart';
+import 'package:seeds/v2/design/app_theme.dart';
+import 'package:seeds/i18n/profile.i18n.dart';
+import 'package:seeds/v2/navigation/navigation_service.dart';
+import 'package:seeds/v2/domain-shared/ui_constants.dart';
+import 'package:seeds/v2/screens/profile_screens/profile/interactor/viewmodels/bloc.dart';
+
+class CitizenshipCard extends StatelessWidget {
+  final ProfileModel profile;
+
+  const CitizenshipCard({
+    Key? key,
+    required this.profile,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return profile.status != ProfileStatus.citizen
```

Comment:
I would have put the BlocBuilder in this widget instead of its parent and here handle all the logic.
```
  Widget build(BuildContext context) {
           return BlocBuilder<ProfileBloc, ProfileState>(
              buildWhen: (previous, current) => previous.profile != current.profile,
              builder: (context, state) {
                return state.profile != null || profile.status != ProfileStatus.citizen? Container(
             ...
             ...
   : const SizedBox.shrink();
```

Reply:
Thank you for the review, I put the logic on the citizen card as suggested